server/controllers/refer/referralsController.test.ts

```diff
@@ -344,4 +344,42 @@ describe('ReferralsController', () => {
       })
     })
   })
+
+  describe('submit', () => {
+    const referral = referralFactory.build({ status: 'referral_started' })
+
+    beforeEach(() => {
+      referralService.getReferral.mockResolvedValue(referral)
+      request.params.referralId = referral.id
+    })
+
+    describe('when the body is incorrect', () => {
+      it('calls flash with an appropiate error and redirects to the check answers page', async () => {
+        request.body.confirm = 'no'
+
+        const requestHandler = referralsController.submit()
+        await requestHandler(request, response, next)
+
+        expect(request.flash).toHaveBeenCalledWith('errors', [
+          {
+            href: '#confirm',
+            text: 'Please confirm that the information you have provided is complete, accurate and up to date',
+          },
+        ])
+        expect(response.redirect).toHaveBeenCalledWith(referPaths.checkAnswers({ referralId: referral.id }))
+      })
+    })
+
+    describe('when the body is correct', () => {
```

Comment:
Put the happy path first but other tests seem to utilise `describe` and think it makes sense in this case, happy to discuss though if needs be.

Reply:
Ah, I think I see what you mean. Typically we weren't nesting the happy path, e.g. `referrals/new` has: 

```js
describe('new'
  it('happy path'
  describe('unhappy path'
    it('does something different'
```

`update` is a little different: since multiple fields can be updated, there's a `describe` block specific to each field (currently only `oasysConfirmed`, but more to follow)

Other than that, I think `complete` is the only one that currently has a happy path in `describe` block

I'm not really bothered which way we do it though!